### PR TITLE
convert_model() legacy extensions fix

### PR DESCRIPTION
**Pull request by popovaan**
Root cause analysis: 
cli_parser unites lists of paths into single string for most of parametes. But 'extensions' are expected to be stored as list, as it may contain non-string parameters came from convert_model(). This bug became visible after convert_model() refactoring https://github.com/openvinotoolkit/openvino/pull/14666 which removed redurant postprocessing of params from cli tool.

Solution: 
Keep extensions list as list and not unite to single string in cli parser.

Ticket: 103867

Code:
* [x]  Comments
* [x]  Code style (PEP8)
* [x]  Transformation generates reshape-able IR - N/A
* [x]  Transformation preserves original framework node names - N/A
* [x]  Transformation preserves tensor names - N/A

Validation:
* [x]  Unit tests
* [x]  Framework operation tests - N/A
* [x]  Transformation tests - N/A
* [x]  Model Optimizer IR Reader check - N/A

Documentation:
* [x]  Supported frameworks operations list - N/A
* [x]  Guide on how to convert the **public** model - N/A
* [x]  User guide update - N/A

### CODE REVIEW

**Review by mvafin — APPROVED** [review]

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py [hunk 1]
```diff
@@ -1,15 +1,21 @@
 # Copyright (C) 2018-2023 Intel Corporation
 # SPDX-License-Identifier: Apache-2.0
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py:3]: please add this test running in the pre-commit. Now it isn't, as I understand, so it can be easily broken:)

Comment (rkazants) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py:3]: `@pytest.mark.precommit` mark should not work

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py [hunk 2]
```diff
@@ -120,3 +126,99 @@ def test_mo_convert_extensions(self, params, ie_device, precision, ir_version,
...
+            ov_model = convert_model(model, extensions=ext_path1)
+            flag, msg = compare_functions(ov_model, create_ref_model_1(), False)
+            assert flag, msg
+
+            ov_model = convert_model(model, extensions=[ext_path1, ext_path2])
+            flag, msg = compare_functions(ov_model, create_ref_model_2(), False)
+            assert flag, msg
+
+            ov_model = convert_model(model, extensions=','.join([ext_path1, ext_path2]))
+            flag, msg = compare_functions(ov_model, create_ref_model_2(), False)
```
Comment (rkazants) [inline tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py:197]: may be it is better to move it into `test_mo_convert_legacy_extensions.py`?

**Review by rkazants — COMMENTED** [review]
one comment is to add this into the pre-commit.

File: tools/mo/openvino/tools/mo/convert_impl.py [hunk 3]
```diff
@@ -405,6 +399,11 @@ def prepare_ir(argv: argparse.Namespace):
     if moc_front_end:
         fallback_reasons = check_fallback(argv)
         if len(fallback_reasons) == 0:
```
Comment (popovaan) [inline tools/mo/openvino/tools/mo/convert_impl.py:401]: @rkazants I suggest to move creating of tmp .pb file here, because otherwise if we fallback to legacy path (due to usage of legacy extensions for example), .pb file is not removed. 
tmp .pb file is not used for legacy path, so I suggest to create it after we checked that fallback not needed.

File: tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py (hunk 2 above)
Comment (popovaan) [reply tests/layer_tests/mo_python_api_tests/test_mo_convert_extensions.py]: Done.

**Comment by rkazants** [comment]
@popovaan, please double-check e2e tests are failing.

File: tools/mo/openvino/tools/mo/convert_impl.py [hunk 4]
```diff
@@ -390,12 +390,6 @@ def prepare_ir(argv: argparse.Namespace):
     # TODO: remove this workaround once new TensorFlow frontend supports non-frozen formats: checkpoint, MetaGraph, and SavedModel
     # Now it converts all TensorFlow formats to the frozen .pb format in case new TensorFlow frontend
     is_tf, _, _, _, _ = deduce_legacy_frontend_by_namespace(argv)
```
Comment (rkazants) [inline tools/mo/openvino/tools/mo/convert_impl.py:392]: @popovaan, please move this comment along with the code.
Please do it this in the separate PR.

**Review by rkazants — APPROVED** [review]
